src/scene.lisp

```diff
@@ -118,6 +118,14 @@
                                        `(,obj (gethash ',obj (objects ,scene)))))
      ,@body))
 
+;; TODO Maybe make a macro that can do the job of either with-scene-objects or with-scene-assets
```

Comment:
or a macro that can be used to create both macros, otherwise the code is essentially just copy-and-pasted.

Reply:
The scene macros as they are don't do all that much. I agree they can and probably should do more.

For this particular case, a macro that takes a list of scenes or objects or both would be good. As for what to name that, well... that's one of the two hard problems in computer science. ;-)